src/main/java/com/extremelyd1/listener/PlayerJoinLeaveListener.java

```diff
@@ -71,6 +71,7 @@ public void onPlayerJoin(PlayerJoinEvent e) {
         Player player = e.getPlayer();
 
         if (game.getState().equals(Game.State.PRE_GAME)) {
+            player.getWorld().setGameRule(GameRule.PLAYERS_SLEEPING_PERCENTAGE, game.getConfig().getSleepPercentage());
```

Comment:
Why would this game rule need updating every time a player joins? Does the config value change in the meantime?

Reply:
i tried to add it when the world is generated, but it doesn't apply it

Reply:
When did you set the game rule initially? On world generation or on world **initialization**? I think the latter should be the proper place to do this.
Either way, the current solution circumvents the problem instead of solving it, which I'm not keen on merging.

Reply:
i tried both